n3fit/src/n3fit/scripts/evolven3fit_new.py

```diff
@@ -0,0 +1,376 @@
+"""
+This module contains the CLI for evolven3fit_new
+"""
+import logging
+import pathlib
+from argparse import ArgumentParser
+import sys
+import numpy as np
+
+from evolven3fit_new import eko_utils, evolve, utils
+
+log = logging.getLogger(__name__)
+
+
+def main():
+    parser = ArgumentParser(
+        description="evolven3fit_new - a script with tools to evolve PDF fits"
+    )
+    subparsers = parser.add_subparsers(title="actions", dest="actions")
+    evolvefit_parser = subparsers.add_parser(
+        "evolve",
+        help="Evolves the fitted PDFs. The q_grid starts at the Q0 given by the theory but the last point is q_fin and its number of points can be specified by q_points. If a path is given for the dump option, the eko will be dumped in that path after the computation. If a path is given for the load option, the eko to be used for the evolution will be loaded from that path. The two options are mutually exclusive.",
+    )
+    eko_parser = subparsers.add_parser(
+        "produce_eko",
+        help="Produce the eko for the specified theory. The q_grid starts at the Q0 given by the theory but the last point is q_fin and its number of points can be specified by q_points. The x_grid starts at x_grid_ini and ends at 1.0 and contains the provided number of points. The eko will be dumped in the provided path.",
+    )
+    evolvefit_parser.add_argument(
+        "configuration_folder",
+        help="Path to the folder containing the (pre-DGLAP) fit result",
+    )
+    parser.add_argument(
+        "-q",
+        "--q_fin",
+        type=float,
+        default=None,
+        help="Final q-value of the evolution",
+    )
+    parser.add_argument(
+        "-p",
+        "--q_points",
+        type=int,
+        default=None,
+        help="Number of q points for the evolution",
+    )
+    parser.add_argument(
+        "-n",
+        "--n_cores",
+        type=int,
+        default=1,
+        help="Number of cores to be used",
+    )
+    evolvefit_parser.add_argument(
+        "-l",
+        "--load",
+        type=pathlib.Path,
+        default=None,
+        help="Path of the EKO to be loaded",
+    )
+    evolvefit_parser.add_argument(
+        "-d",
+        "--dump",
+        type=pathlib.Path,
+        default=None,
+        help="Path where the EKO is dumped (optional)",
+    )
+    evolvefit_parser.add_argument(
+        "-f",
+        "--force",
+        action="store_true",
+        help="Force the evolution to be done even if it has already been done",
+    )
+    eko_parser.add_argument(
+        "theoryID", type=int, help="ID of the theory used to produce the eko"
```

Comment:
I'd prefer `theory-id`, because it is more consistent naming scheme for CLI arguments and corresponding Python identifier (that would be `theory_id`). But I know this is something that is consistently used in vp...

Reply:
Yeah so maybe it is better to wait for @RoyStegeman or @Zaharid opinion.

Reply:
Or even @scarlehoff of course

Reply:
Let's use theoryID since as you say that's how it is called everywhere else.

But I have to say that I'm _never_ able to write `theoryID` correctly in less than five tries. So if you change it to anything else I won't complain.